source/binding/Statements.cpp

```diff
@@ -1239,6 +1329,25 @@ ER ForeachLoopStatement::evalRecursive(EvalContext& context, span<const Constant
     return ER::Success;
 }
 
+void ForeachLoopStatement::serializeTo(ASTSerializer& serializer) const {
+    serializer.write("arrayRef", arrayRef);
+
+    serializer.startArray("loopRanges");
+    for (auto const& r : loopRanges) {
+        serializer.write("range", r.toString());
```

Comment:
Seems like the same problem here, adding properties to an array. It seems my ASTSerializer API is too confusing :(

Reply:
How can I generate a list of strings? Can I just add a function like `serializer.serialize("str")` and use it inside the loop?

Reply:
Yeah, that's how I would do it.

Reply:
Perhaps the existing write() methods should have been named writeProperty() to make it clear what they're doing. I might make that change in a different diff, I'll think about it some more.